docs for dashboard (#253)
hubverse-org/hubDocs: docs/source/user-guide/dashboards.md

Below, we describe how to set up each of these three components of the dashboard system and the use of GitHub actions for building the site.

## Quickstart -- building a dashboard

The Hubverse provides a [template dashboard repository](https://github.com/hubverse-org/hub-dashboard-template) to facilitate the process of setting up a dashboard for a modeling hub. Use this template to create a dashboard repository for your hub by clicking the "Use this template" button near the top right of the GitHub page for the template:

```{figure} ../images/hub-dashboard-template-use-template.png
---
figclass: margin-caption
alt: A screenshot of the GitHub repository for the hub dashboard template. There is a red circle around the "Use this template" button.
name: hub-data-relations
---
Creating a hub dashboard from the template.
```

Then do the following to customize your dashboard (see the sections below for more detail):

1. Add markdown content to `pages/`

> 2. Update `site-config.yml`

Reviewer: Below you link to `[YAML format](https://en.wikipedia.org/wiki/YAML)`, but maybe we should put a note at the top saying all `.yml` files indicate that format? Then you can remove it from below ("This file is written in the [YAML format](https://en.wikipedia.org/wiki/YAML).")

Author: Resolving this since Zhian added the YAML link above.

    1. `hub` is the github slug for your active hub. This example defaults to the CDC FluSight hub
    2. `title` is the title of your dashboard

>     3. `pages` is a list of pages you want included in the top bar. These will be shown after the home page (index.html) and the visualizaton and evaluation pages if they are included (see the next points).

Reviewer: Is `pages` optional? I don't see it in https://github.com/reichlab/flusight-dashboard/blob/main/site-config.yml , for example. If not, is the default "all pages?"

name: dashboard-viz-screenshot
---
A screenshot of the visualization in the FluSight dashboard.
```

### Configuring the PredTimeChart visualization module

If you don't want to include a visualization using the PredTimeChart module in your dashboard, delete the `predtimechart-config.yml` file from your dashboard repository.

To include the PredTimeChart visualization, edit the `predtimechart-config.yml` file to match your hub. You can view the [raw schema](https://raw.githubusercontent.com/hubverse-org/hub-dashboard-predtimechart/main/src/hub_predtimechart/ptc_schema.py) for this file to see the detailed specification of its contents. Below, we give an example file based on the [FluSight forecast hub](https://github.com/cdcepi/FluSight-forecast-hub) and describe the fields to include:

```yaml
---
rounds_idx: 0
model_tasks_idx: 1
reference_date_col_name: 'reference_date'
target_date_col_name: 'target_end_date'
horizon_col_name: 'horizon'
initial_checked_models: ['FluSight-ensemble', 'FluSight-baseline']

> target_data_file_name: 'target-hospital-admissions.csv'

Author: ```suggestion
```

Author: finishing the job of removing references to the `target_data_file_name` field

disclaimer: 'Be careful when interpreting these forecasts.'
task_id_text:
  location:
    "US": "United States"
    "01": "Alabama"
    "02": "Alaska"
    "04": "Arizona"
    "05": "Arkansas"
    "06": "California"
```

This file is written in the [YAML format](https://en.wikipedia.org/wiki/YAML), and it contains the following fields:

 - `rounds_idx`: The 0-based index of the `rounds` entry in the hub's `tasks.json` configuration file to use for the visualization.
    - As is described in more detail in the limitations section below, the visualization is currently only able to show predictions for defined within one block of the `rounds` section of the `tasks.json` configuration.
 - `model_tasks_idx`: The 0-based index of the `model_tasks` entry under `rounds_idx` to use for the visualization.

>     - As is described in more detail in the limitations section below, the visualization is currently only able to show predictions that are defined in a single "model tasks block".

Reviewer: "only able to show predictions for __ defined within one block" - I think there's a word omitted from the sentence. Maybe it should read "only able to show predictions <that are defined> within one block" ?

>     - As is described in more detail in the limitations section below, the visualization is currently only able to show predictions that are defined in a single "model tasks block".

Author: ```suggestion
    - As is described in more detail in the limitations section below, the visualization is currently only able to show predictions for tasks that are defined in a single "model tasks block".
```

## PredEvals evaluation (optional)

The PredEvals module creates an interactive display of scores for predictions. Dashboard users can view overall scores in a table, or see line plots or heatmaps visualizing scores broken down by a task id variable.

```{figure} ../images/dashboard-eval-table.png
---
figclass: margin-caption
name: dashboard-eval-table
---
A screenshot of the evaluation table in the COVID-19 Forecast Hub dashboard. The table shows overall scores for each model.
```

```{figure} ../images/dashboard-eval-lineplot.png
---
figclass: margin-caption
name: dashboard-eval-lineplot
---
A screenshot of the COVID-19 Forecast Hub dashboard showing a line plot of relative WIS values for models, broken down by the target end date of the prediction.
```

```{figure} ../images/dashboard-eval-heatmap.png
---
figclass: margin-caption
name: dashboard-eval-heatmap
---
A screenshot of the COVID-19 Forecast Hub dashboard showing a heatmap of relative WIS values for models, broken down by the location of the prediction.

> ```

Author: It's not that great to just dump these screenshots here, but I tried using the [subfigures module for sphinx](https://sphinx-subfigure.readthedocs.io/en/latest/#usage) to arrange them more nicely and failed -- gave up after an hour of fiddling.

:::{admonition} Redundancy of `round_filters` and `task_filters`
:class: note

In principle, it would be possible to implement `round_filters` by specifying the set of rounds to include within the `task_filters` block. While this is possible, specifying `round_filters` is more convenient and maintainable, so we recommend using those settings where appropriate.

:::

### PredEvals limitations and requirements

The PredEvals module has several important limitations:

 - The evaluation tool only supports hubs that have a single entry in the `rounds` section of the `tasks.json` configuration file.
 - Only hubs with `round_id_from_variable` set to `true` in the `tasks.json` configuration file are supported.
 - Supported output types include `mean`, `median`, `quantile`, and `pmf`. However, support for ordinal pmf predictions is still experimental. The `sample` and `cdf` output types are not supported.

## Using GitHub actions to build site contents and data

> The template dashboard repository comes with a set of GitHub workflows that can be used to build the dashboard site contents either according to a recurring schedule or on an as-needed basis by hub administrators.

Reviewer 2: ```suggestion
The template dashboard repository comes with a two GitHub workflows that are responsible for building the site contents and data. These are located in the `.github/workflows/` folder: 

- `build-site.yml` is used to build the dashboard site contents every time a change happens on the `main` branch of the dashboard repository. 
- `build-data.yml` runs weekly on Thursdays at 5:45PM UTC to update the underlying data for the dashboards. This schedule is configurable to work with your hub. Note that this does not run on every commit.

You can also [manually run](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow#running-a-workflow) either of these workflows if you need to rebuild your site or data.

If the above defaults looks good to you, then as long as you've enabled GitHub Actions, then there's nothing you need to do!
```

> TODO: add information about how to do these things.

Author: @zkamvar can you add some stuff here?

:::{admonition} Redundancy of `round_filters` and `task_filters`
:class: note

In principle, it would be possible to implement `round_filters` by specifying the set of rounds to include within the `task_filters` block. While this is possible, specifying `round_filters` is more convenient and maintainable, so we recommend using those settings where appropriate.

:::

### PredEvals limitations and requirements

The PredEvals module has several important limitations:

 - The evaluation tool only supports hubs that have a single entry in the `rounds` section of the `tasks.json` configuration file.
 - Only hubs with `round_id_from_variable` set to `true` in the `tasks.json` configuration file are supported.
 - Supported output types include `mean`, `median`, `quantile`, and `pmf`. However, support for ordinal pmf predictions is still experimental. The `sample` and `cdf` output types are not supported.

## Using GitHub actions to build site contents and data

> The template dashboard repository comes with a two GitHub workflows that are responsible for building the site contents and data. These are located in the `.github/workflows/` folder: 

Author: ```suggestion
The template dashboard repository comes with two GitHub workflows that are responsible for building the site contents and data. These are located in the `.github/workflows/` folder: 
```
